fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateMaterializedViewStmt.java

```diff
@@ -517,7 +515,7 @@ private static void analyzeOrderByClause(CreateMaterializedViewStmt statement,
             throw new SemanticException("The number of columns in order clause must be less then " + "the number of "
                     + "columns in select clause");
         }
-        if (beginIndexOfAggregation != -1 && (orderByElements.size() != (beginIndexOfAggregation))) {
+        if (selectRelation.getGroupBy().size() != orderByElements.size()) {
```

Comment:
Check the size of `group by` and `order by` can't fix all situation? We should check the column id instead.